File: Java-Module/src/main/java/com/sim/javamodule/domain/user/service/dto/UserInfoResponse.java
package com.sim.javamodule.domain.user.service.dto;

import com.sim.javamodule.domain.user.domain.User;
import lombok.Getter;

@Getter
public record UserInfoResponse(Long id, String username, String password) {
    public static UserInfoResponse mapUserToUserInfoResponse(final User user){
        return new UserInfoResponse(user.getId(), user.getUsername(), user.getPassword());
    }
}

File: Kotlin-Module/src/main/kotlin/com/sim/kotlin/domain/user/service/dto/UserInfoResponse.kt
package com.sim.kotlin.domain.user.service.dto

import com.sim.kotlin.domain.user.domain.User

data class UserInfoResponse(
    val id: Long?,
    val username: String,
    val password: String,
){
    companion object{
        fun mapUserToUserInfoResponse(user: User)=UserInfoResponse(user.id, user.username, user.password)
    }
}